### ISSUE TITLE
Suggestion: You may be able to use MIT-licensed code only

### ISSUE THREAD

**Issue by Warr1024** [Irevol/minetest-citadel#28]
I noticed that your citadel_core mod, where most of the real work is done, is using MIT-licensed code, but you're forced to use LGPL because of some 3rd party mods.

You may be able to remove the other mods and license your code under pure MIT, which also makes the licensing/distribution for the game overall a bit simpler.

- The ambient light mod does something pretty trivial and precedents exist in at least Piranesi Restoration for fully MIT-licensed ambient lighting.  It's also pretty trivial to reimplement anyway.
- sfinv you might not need if you simplify inventory, and you may be best served by a simpler fully custom formspec anyway once you've got the guidance moved out of it.
- I think you're using xpanes *just* for the bar doors; those are not hard to replace with a simple nodebox model.
- I don't see what you're using player_api for; there doesn't even seem to be any player model.  If you want one, you can use the NodeCore MIT-licensed player model, textures, and code, as used in Klots and Piranesi Restoration.

**Comment by andyca111100** [comment]
Agree. I was also confused when I saw mods from MTG in Citadel, which weren't even trimmed down (for example, sounds from `xpanes` are not used, I think?)
Also, what de desktop.ini doin' in the `citadel_core/textures`?

Update: `star.png` and `default_leaves.png` are clearly from MTG, first one being `bones_front.png` from `bones` mod, and second one is, well, `default_leaves.png` from `default`. Replacing those textures and (re)licensing others under CC0 would be also great. I recommend reading [a post on Minetest](https://forum.minetest.net/viewtopic.php?t=27881) on this topic.

After all, as Warr said on Discord:

> ***####ing textures are licensed under CC-BY-SA, non-####ing under CC0.***

He is a smart dude, I believe.

**Comment by andyca111100** [comment]
Update 2: Also, those textures are 16x16, while others are 32x32, and they don't match the style. And `star.png` is a confusing easter egg, at least to me. And it might be too late, but doesn't they [MTG textures] break the "Everything from scratch, except libraries" Game Jam rule?

**Comment by Irevol** [comment]
`star.png` was never intended to be an easter egg or be in the game; it was just to help me design puzzles. `default_leaves.png` wasn't meant to be permanent either. Somehow I forgot about it. I'd be surprised if I was punished for those 2 textures.
<br> I also get the feeling that Warr is indeed a smart dude and will change the license to CC0.

**Comment by andyca111100** [comment]
That's great! After replacing mods, Citadel will become MIT/CC0 instead of more restrictive LGPL/CC BY-SA. But you can only re-license your own textures, so redoing redrawing those included in the mods that will be removed is also necessary. And don't forget about other media!.. I hope I didn't sound impudent (or something like that, I'm using a translator) and Warr won't get mad at me for doing important stuff sillily?..

**Comment by andyca111100** [comment]
Update: the Totem is based upon Minetest logo, which is under CC BY-SA 3.0... uh-oh. Perhaps removing or remaking it is a good idea?? I didn't like it after all (because of the spot where it is hidden, when I first saw it I thought it was a bug, because "You can't teleport into walls"), and the texture is a bit mid if you look closely... and also the same situation as with leaves? 

![зображення](https://github.com/Irevol/minetest-citadel/assets/113429882/909dfc32-afc1-413f-91e3-983658896772)

Perhaps I should be sorry for the inconviniences that I've created...
